web/src/components/thread/ThreadMenu/useThreadMenu.ts

```diff
@@ -66,8 +67,15 @@ export function useThreadMenu({
   async function handleDelete() {
     await handle(
       async () => {
-        await deleteThread(thread.id);
-        router.push("/");
+        await withUndo({
+          message: "Thread deleted",
+          duration: 5000,
+          action: async () => {
+            await deleteThread(thread.id);
+            router.push("/");
```

Comment:
This push to the index page should only happen if you're on the thread page itself.

Or maybe not, there's actually a "Deleted" state on the thread screen, it doesn't become inaccessible necessarily but depends on the member's access level. I think "MANAGE_THREADS" permission allows you to load deleted threads. But I'll have to check on that.

The issue with the redirect is, if you delete a thread from within a category page or another non-index page, you get unexpectedly redirected for no reason.